extensions/cl_ext_image_requirements_info.asciidoc

```diff
@@ -0,0 +1,389 @@
+// Copyright 2018-2021 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+include::{generated}/api/api-dictionary-no-links.asciidoc[]
+:source-highlighter: coderay
+
+= cl_ext_image_requirements_info
+:R: pass:q,r[^(R)^]
+Khronos{R} OpenCL Working Group
+
+== Name Strings
+
+`cl_ext_image_requirements_info`
+
+== Contact
+
+Please see the *Issues* list in the Khronos *OpenCL-Docs* repository: +
+https://github.com/KhronosGroup/OpenCL-Docs
+
+== Contributors
+
+Kevin Petit, Arm Ltd. +
+Jeremy Kemp, Imagination Technologies +
+Alastair Murray, Codeplay Software Ltd. +
+Balaji Calidas, Qualcomm +
+
+== Notice
+
+include::../copyrights.txt[]
+
+== Status
+
+Draft spec, NOT APPROVED!!
+
+== Version
+
+Built On: {docdate} +
+Version: 0.5.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification Version 3.0.9.
+
+This extension requires OpenCL 3.0.
+
+== Overview
+
+This extension enables applications to query requirements for an image without
+having to create the image.
+
+== New API Functions
+
+[source,c]
+----
+cl_int clGetImageRequirementsInfoEXT(
+    cl_context context,
+    const cl_mem_properties* properties,
+    cl_mem_flags flags,
+    const cl_image_format* image_format,
+    const cl_image_desc* image_desc,
+    cl_image_requirements_info_ext param_name,
+    size_t param_value_size,
+    void* param_value,
+    size_t* param_value_size_ret);
+----
+
+== New API Types
+
+[source,c]
+----
+typedef cl_uint cl_image_requirements_info_ext;
+----
+
+== New API Enums
+
+Accepted value for the _param_name_ parameter to *clGetImageRequirementsInfoEXT*:
+
+[source,c]
+----
+CL_IMAGE_REQUIREMENTS_ROW_PITCH_ALIGNMENT_EXT    0x1290
+CL_IMAGE_REQUIREMENTS_BASE_ADDRESS_ALIGNMENT_EXT 0x1292
+CL_IMAGE_REQUIREMENTS_SIZE_EXT                   0x12B2
+CL_IMAGE_REQUIREMENTS_MAX_WIDTH_EXT              0x12B3
+CL_IMAGE_REQUIREMENTS_MAX_HEIGHT_EXT             0x12B4
+CL_IMAGE_REQUIREMENTS_MAX_DEPTH_EXT              0x12B5
+CL_IMAGE_REQUIREMENTS_MAX_ARRAY_SIZE_EXT         0x12B6
+----
+
+== Modifications to the OpenCL API Specification
+Modify Section 5.3.1, *Creating Image Objects*) ::
```

Comment:
Editorial: the indentation is a little weird here, and there's a mis-matched parenthesis.